scripts/check-selectors.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+# Set -u to error out if we use an unset variable.
+# Set -o pipefail to propagate errors in a pipeline.
+# Set -e to exit on any non-zero exit code.
+set -uo pipefail
+
+# Collection of repositories to check.
+declare -A repositories
+
+# is_passing is a boolean variable that is set to true if the check passes.
+is_passing=true
+
+# Define colors.
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# External dependencies: repositories, is_passing.
+function check() {
```

Comment:
just fyi `function` is superfluous when you already use `()` 
```suggestion
check() {
```

see: https://tldp.org/HOWTO/Bash-Prog-Intro-HOWTO-8.html

Reply:
I also think this function can receive the repositories are argument.  Here is something I tried which you may want to consider. 

```
#!/usr/bin/env bash

set -eu -o pipefail

warn() {
	echo "WARN: $*"
}

check_repositories() {
	# Use "$@" to pass the entire array as an argument
	local -n repos=$1

	for repo_url in "${!repos[@]}"; do

		echo "repo_url: $repo_url"

		## assert it is an array
		# set -x
		[[ "${repos[$repo_url]}" =~ ^\(.* ]] || {
			warn "invalid repo url $repo_url -> ${repos[$repo_url]}"
			continue # or return 1
		}

		git ls-remote "$repo_url" &>/dev/null || {
			warn "invalid repo url: $repo_url"
			continue # or return 1
		}

		local -a metrics="${repos[$repo_url]}"

		for metric in ${metrics[@]}; do
			echo "    metric: $metric"
		done
	done
}

make_array() {
	echo "($@)"
}

main() {

	declare -A repo_metrics=(
		["foo"]="$(make_array "abc" "def")"
		["bar"]="$(make_array "xyz" "mno" "foobar")"
		["invalid"]="xyz"
		["valid"]="$(make_array "too" "moo")"
		["https://github.com/kubernetes/kube-state-metrics"]="$(make_array internal/store kube_ kubeStateMetricsSelector)"
	)

	check_repositories repo_metrics
}

main "$@"
```